### PR TITLE
[RPD-284] Update AzureRunner to select the Terraform files corresponding to the stack name in the config file

### PR DESCRIPTION
**Pull request by Christopher-Norman**
This PR:
1. Renames the `resources` Terraforms file to `default`.
2. Adds functionality within `core.py` provision function to collect the stack name and use that as a pointer to the specified set of Terraform files.
3. If this value does not exist it will create it and use the `default` stack.

This PR also fixes a bug where an error would be thrown due to the `matcha.config.json` file not being empty but not containing information relating to a resource group.

This has been tested with both stacks. 

One possible improvement is that when using the `default` stack an empty `chroma` folder is still created within `.matcha/infrastructure/resources`.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [x] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

**Review by KirsoppJ — APPROVED** [review]
LGTM

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice, a couple of architectural changes and we're good to go

File: src/matcha_ml/config/matcha_config.py [hunk 1]
```diff
@@ -126,6 +126,30 @@ def from_dict(state_dict: Dict[str, Dict[str, str]]) -> "MatchaConfig":
...
+        Returns:
+            str: The name of the current stack being used as a string.
+        """
+        try:
+            stack_name = (
+                MatchaConfigService.read_matcha_config()
+                .find_component("stack")
+                .find_property("name")
+            ).value
+        except MatchaError:
```
[the remark this answers is not in the document]
Comment (Christopher-Norman) [reply src/matcha_ml/config/matcha_config.py:142]: I think this is fine, defaulting to a `default` stack makes sense and then the config is updated to reflect this default.

[the remark this answers is not in the document]
Comment (JonoCX) [reply src/matcha_ml/config/matcha_config.py:142]: I agree, it makes sense to default to the default stack when a stack hasn't been set. But from an architectural standpoint, setting the stack within a function that gets the stack doesn't quite make sense as it's mixed functionality. 

I'd be included to return `None` here if no stack can be found, which is a signal to the `provision` function in core to call the `stack_set` function with `stack_name='default'`, e.g.,

```python
# logic in core.provision
if service.get_stack() is None: 
    stack_set(stack_name='default')

...

os.path.join(
    os.path.dirname(__file__),
    os.pardir,
    "infrastructure",
    service.get_stack().value,
)
```

**Review by JonoCX — APPROVED** [review]
LGTM. It'd be good to address the nit to simplify the code a little

File: src/matcha_ml/core/core.py [hunk 2]
```diff
@@ -271,21 +271,27 @@ def provision(
...
 
-    stack_name = MatchaConfigService.get_current_stack_name()
-
     with remote_state_manager.use_lock(), remote_state_manager.use_remote_state():
         project_directory = os.getcwd()
         destination = os.path.join(
             project_directory, ".matcha", "infrastructure", "resources"
         )
+
+        stack = MatchaConfigService.get_stack()
```
[the remark this answers is not in the document]
Comment (Christopher-Norman) [reply src/matcha_ml/core/core.py:286]: Also, I believe `lower()` is required as the values are stored in `matcha.config.json` as `LLM` or `DEFAULT`

[the remark this answers is not in the document]
Comment (JonoCX) [reply src/matcha_ml/core/core.py:286]: Perhaps adding a type to the `stack` variable in the original implementation may help: `stack: Optional[MatchaConfigComponentProperty] = ...`

On the `lower()`, if the internals are using the stack enum (they should), then the values should all be lower case 

No worries if there's no work around for `mypy`

[the remark this answers is not in the document]
Comment (Christopher-Norman) [reply src/matcha_ml/core/core.py:286]: Removed lower casing. No work around for `mypy`.

**Review by JonoCX — APPROVED** [review]
✅ 🌯

**Pull request merged**